runtime/runtime-standalone/src/lib.rs

```diff
@@ -0,0 +1,394 @@
+use near_pool::{types::PoolIterator, TransactionPool};
+use near_primitives::{
+    account::Account,
+    errors::RuntimeError,
+    hash::CryptoHash,
+    receipt::Receipt,
+    state_record::StateRecord,
+    transaction::{ExecutionOutcome, ExecutionStatus, SignedTransaction},
+    types::AccountId,
+    types::{Balance, BlockHeight, EpochHeight, Gas, StateChangeCause},
+};
+use near_runtime_configs::RuntimeConfig;
+use near_store::{get_account, set_account, Store, Trie, TrieUpdate};
+use node_runtime::{state_viewer::TrieViewer, ApplyState, Runtime};
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[derive(Debug, Default)]
+pub struct GenesisConfig {
+    pub genesis_time: u64,
+    pub gas_price: Balance,
+    pub gas_limit: Gas,
+    pub runtime_config: RuntimeConfig,
+    pub state_records: Vec<StateRecord>,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct Block {
+    prev_block: Option<Box<Block>>,
+    state_root: CryptoHash,
+    gas_burnt: Gas,
+    pub epoch_height: EpochHeight,
+    pub block_height: BlockHeight,
+    pub block_timestamp: u64,
+    pub gas_price: Balance,
+    pub gas_limit: Gas,
+}
+
+impl Block {
+    pub fn genesis(genesis_config: &GenesisConfig) -> Self {
+        Self {
+            prev_block: None,
+            state_root: CryptoHash::default(),
+            block_height: 1,
+            epoch_height: 1,
+            block_timestamp: genesis_config.genesis_time,
+            gas_price: genesis_config.gas_price,
+            gas_limit: genesis_config.gas_limit,
+            gas_burnt: 0,
+        }
+    }
+
+    pub fn produce(&self, new_state_root: CryptoHash) -> Block {
+        Self {
+            gas_price: self.gas_price,
+            gas_limit: self.gas_limit,
+            block_timestamp: self.block_timestamp + 1,
+            prev_block: Some(Box::new(self.clone())),
+            state_root: new_state_root,
+            block_height: self.block_height + 1,
+            epoch_height: self.epoch_height + 1, // TODO: simulate epoch_height
```

Comment:
it shouldn't be too hard to simulate. Just add `epoch_length` in `GenesisConfig` and then calculate the epoch switch accordingly.

Reply:
Yeah, but let's keep it as is for now. This implementation is already useful.

Reply:
Maid it like `epoch_height: self.block_timestamp + 1 / epoch_length,`